adapters/sinatra/lib/opentelemetry/adapters/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      module Middlewares
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            tracer.in_span(
+              env['PATH_INFO'],
+              attributes: { 'component' => 'http',
+                            'http.method' => env['REQUEST_METHOD'],
+                            'http.url' => env['PATH_INFO'] },
+              kind: :server,
+              with_parent_context: parent_context(env)
+            ) do |span|
+              app.call(env).tap { |resp| trace_response(span, env, resp) }
+            end
+          end
+
+          private
+
+          attr_reader :app
+
+          def parent_context(env)
+            OpenTelemetry::Adapters::Sinatra::Adapter.propagator.extract(env)
```

Comment:
Is it the job of a propagator to inspect this payload and decide whether it is trustworthy? Can someone craft headers malicious headers that would cause confusion in tracing systems?  Maybe a simple flag to always start a new trace instead of accepting existing context.  Or, this might be a case where offering hooks into the instrumentation would enable more sophisticated behavior like client code looking at the request and deciding whether to allow the context it contains to be extracted